cmd/kubeadm/app/cmd/init.go

```diff
@@ -269,6 +270,11 @@ func (i *Init) Run(out io.Writer) error {
 		return fmt.Errorf("couldn't parse kubernetes version %q: %v", i.cfg.KubernetesVersion, err)
 	}
 
+	// Upload currently used configuration to the cluster
+	if err := saveconfigphase.SaveConfiguration(i.cfg, client); err != nil {
```

Comment:
To me, "upload" is just as vague as "save," since it isn't obvious what happens if the data already exists. We already have the pattern of explicitly saying "CreateOrUpdate" elsewhere in our phases, so I'm wondering why we don't want to follow that. If we're trying to have kubeadm taken seriously as a library / toolkit, I think having a consistent API experience is pretty important.

But, three different people had three different ideas for the naming in this PR :-). Since this is basically introducing a new kubeadm API, maybe we should cast the net a little wider and get a little bit of consensus on how to name it?

Or, if we already have precedent of "all of the phases are alpha anyway, so we can rename them whenever we want," maybe it's not so important to get it right the first time. I haven't been involved in the kubeadm adoption working group, so I'm not sure what the sentiment is around using these phases.